### PR TITLE
Retry wait command.

### PR DESCRIPTION
**Pull request by lkysow**
This was failing on AKS occasionally.

See https://app.circleci.com/pipelines/github/hashicorp/consul-helm/1866/workflows/809b64f2-cce9-479a-a1c6-42ae11defba3/jobs/4844/steps

```
 command.go:100: 2020-11-20T00:48:35Z Running command kubectl with args [--kubeconfig /home/circleci/.kube/consul-k8s-1376936457 --namespace default wait --for=condition=available --timeout=1m deploy/static-server]
    command.go:179: 2020-11-20T00:48:47Z The connection to the server consul-k8s-1376936457-8fc76f5b.hcp.westus2.azmk8s.io:443 was refused - did you specify the right host or port?
    kubectl.go:82: 
        	Error Trace:	kubectl.go:82
        	            				deploy.go:67
        	            				health_checks_test.go:59
        	Error:      	Received unexpected error:
        	            	error while running command: exit status 1; The connection to the server consul-k8s-1376936457-8fc76f5b.hcp.westus2.azmk8s.io:443 was refused - did you specify the right host or port?
        	Test:       	TestHealthChecks/secure:_true,_auto-encrypt:_false
```

I don't know why `kubectl apply` was succeeding and `kubectl wait` wasn't, so this may not solve the problem if all kubectl commands can randomly fail.

### CODE REVIEW

File: test/acceptance/framework/k8s/deploy.go [hunk 1]
```diff
@@ -64,7 +64,19 @@ func DeployKustomize(t *testing.T, options *k8s.KubectlOptions, noCleanupOnFailu
...
-	RunKubectl(t, options, "wait", "--for=condition=available", "--timeout=1m", fmt.Sprintf("deploy/%s", deployment.Name))
+	// This command on AKS often gave us the error:
+	// "The connection to the server consul-k8s-1376936457-8fc76f5b.hcp.westus2.azmk8s.io:443 was refused - did you specify the right host or port?"
+	// So now we retry up to 3 times.
+	retrier := &retry.Counter{Count: 3}
+	retry.RunWith(retrier, t, func(r *retry.R) {
+		// NOTE: It's okay to use `t` here instead of `r` because `t` is only
+		// used to error on command construction. If the command itself fails
+		// then it will return an error which we correctly use `r` with.
+		_, err := RunKubectlAndGetOutputE(t, options, "wait", "--for=condition=available", "--timeout=1m", fmt.Sprintf("deploy/%s", deployment.Name))
```
Comment (lkysow) [inline test/acceptance/framework/k8s/deploy.go:75]: Had to do this instead of `RunKubectl(t, ...)`  because `r` doesn't implement the necessary interface.

**Review by thisisnotashwin — APPROVED** [review]
here is hoping that problem goes away 🤞

**Review by ndhanushkodi — APPROVED** [review]
🎉 🤞

**Comment by lkysow** [comment]
Closed by #703

**Pull request closed without merging**